python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -3205,7 +3205,7 @@ def _get_partition_data_versions(
                     == DagsterEventType.ASSET_OBSERVATION.value,
                 ),
                 SqlEventLogStorageTable.c.asset_key == asset_key.to_string(),
-                SqlEventLogStorageTable.c.partition.in_(partitions),
+                SqlEventLogStorageTable.c.partition.in_(partitions) if partitions else True,
```

Comment:
nit: I would rather you just be explicit with an if statement to include this clause on the query; doing the True passthrough... it's a bit too tricksy

Reply:
it is neat tho I will admit